proxy/envoy/route.go

```diff
@@ -142,48 +143,42 @@ func buildHTTPRoute(rule *proxyconfig.RouteRule, port *model.Port) *HTTPRoute {
 		}
 	}
 
+	destination := service.Hostname
+
 	if len(rule.Route) > 0 {
-		clusters := make([]*WeightedClusterEntry, 0)
+		route.WeightedClusters = &WeightedCluster{}
 		for _, dst := range rule.Route {
-			destination := dst.Destination
-
-			// fallback to rule destination
-			if destination == "" {
-				destination = rule.Destination
-			}
+			// TODO: add destination service override from the route weights
 
```

Comment:
Can you please file a P0 issue or even a bug so that we don't forget to fix this before release?

Reply:
Will do.